### PR TITLE
[SYCL] Bump the version 6.2.0->7.0.0 and update the versioning section in docs

### PR DESCRIPTION
**Pull request by aelovikov-intel**

### CODE REVIEW

File: sycl/doc/developer/ABIPolicyGuide.md [hunk 1]
```diff
@@ -35,13 +35,12 @@ Adding a new exported symbol is considered to be non-breaking change.
...
-uplift `V`. It is pull request author responsibility to accordingly update
-`V` part of library version. If `V > 0` on release date, then `MAJOR_VERSION`
-is uplifted and `V` is set to zero. `MAJOR`, `MINOR`, and `PATCH` versions are
-not updated between releases.
+version `X-1`. `MINOR` indicates a non-breaking change. We only bump each of
+them once per development cycles between releases. Immediately after the release
+branch has been brached off, the `MINOR` version is incremented. If an
+ABI-breaking change is introduced then it should be accompanied with the bump of
+a `MAJOR` version in the same PR (unless it had been bumped before in this
+development cycle).
```
Comment (steffenlarsen) [inline sycl/doc/developer/ABIPolicyGuide.md:43]: When would "unless it had been bumped before in this development cycle" ever apply?

**Review by steffenlarsen — APPROVED** [review]
LGTM!

**Comment by pvchupin** [comment]
@aelovikov-intel should we un-draft and finalize this?

**Comment by aelovikov-intel** [comment]
We need to discuss (probably offline via a call) on how to deal with PATCH/DEV increments, i.e. if we want them or not.

**Review by steffenlarsen — APPROVED** [review]
🚀

**Review by stdale-intel — APPROVED** [review]
LGTM

**Comment by aelovikov-intel** [comment]
CUDA's `SYCL :: Regression/same_unnamed_kernels.cpp` is known - https://github.com/intel/llvm/issues/9174.

**Comment by aelovikov-intel** [comment]
L0 GEN9 Linux seems to be stuck due to infrastructure problems (1.5 hour on "Set up job" stage). I think we can merge this in because that task passed on a previous revision of this PR (https://github.com/intel/llvm/actions/runs/4787697483).

**Pull request merged**